### DISCUSSION TITLE
REST: Querying with filter based on Many to Many?

### DISCUSSION THREAD

**Discussion by oguruma1218** [supabase/supabase#14411] [Questions]
I have a table called Products, a table called Attributes, and a table called AttributeValues and a join table called ProductsToAttributeValues.

Products:
 product_id 
 name (eg "A Red Book")

Attributes
 attribute_id
 name (eg "Color")

AttributeValues
 attributevalue_id 
 value (eg "Red")

ProductsToAttributeValues
 productstoattributevalues_id
 attribute_id
 attributevalue_id

I'd like to use REST to find all of the Products that are "Color = Red"

Can this be done with a native REST call?

**Comment by GaryAustin1** [comment]
This should be possible.
You can filter on embedded relationships and the REST API (PostgREST) will figure out relationships thru the connected tables.
Your many to many table needs to have both foreign keys as part of the primary key for the API to process the relationships thru that.
You can use the !inner tag after a table if you want the filter on that table to filter the "parent" table.

https://postgrest.org/en/stable/references/api/resource_embedding.html#resource-embedding
